src/univariate/continuous/generalizedgaussian.jl

```diff
@@ -0,0 +1,107 @@
+
+```julia
+GeneralizedGaussian()           # GGD with shape 2, scale 1, location 0, (the Normal distribution)
+GeneralizedGaussian(loc,s,sh)   # GGD with location loc, scale s, and shape sh
+
+params(d)                       # Get the parameters, i.e. (loc,s,sh,)
+shape(d)                        # Get the shape parameter, sh
+scale(d)                        # Get the scale parameter, s
+location(d)                     # Get the location parameter, loc
+```
+
+External Links
+ * [Generalized Gaussian on Wikipedia](http://en.wikipedia.org/wiki/Generalized_normal_distribution)
+ """
+
+struct GeneralizedGaussian{T<:Real} <: ContinuousUnivariateDistribution
+    α::T
+    μ::T
+    β::T
+
+    GeneralizedGaussian{T}(μ::T,α::T,β::T) where {T} = (@check_args(GeneralizedGaussian, α > zero(α) && β > zero(β)); new{T}(μ,α,β))
+end
+
+GeneralizedGaussian(μ::T,α::T,β::T) where {T<:Real} = GeneralizedGaussian{T}(μ,α,β)
+GeneralizedGaussian(μ::Integer,α::Integer,β::Integer) = GeneralizedGaussian(Float64(μ),Float64(α),Float64(β))
+GeneralizedGaussian(β::Float64) = GeneralizedGaussian(0.0, 1.0, β)
+GeneralizedGaussian() = GeneralizedGaussian(0.0, 1.4142, 2.0) # approximate scale with unity std deviation and shape 2
+
+#### Conversions
+
+convert(::Type{GeneralizedGaussian{T}}, μ::S, α::S, β::S) where {T <: Real, S <: Real} = GeneralizedGaussian(T(μ),T(α),T(β))
+convert(::Type{GeneralizedGaussian{T}}, d::GeneralizedGaussian{S}) where {T <: Real, S <: Real} = GeneralizedGaussian(T(d.μ), T(d.α), T(d.β))
+
+@distr_support GeneralizedGaussian -Inf Inf
+
+
+#### Parameters
+params(d::GeneralizedGaussian) = (d.μ, d.α, d.β)
+@inline partype(d::GeneralizedGaussian{T}) where {T<:Real} = T
+
+location(d::GeneralizedGaussian) = d.μ
+shape(d::GeneralizedGaussian) = d.β
+scale(d::GeneralizedGaussian) = d.α
+
+
+#### Statistics
+
+mean(d::GeneralizedGaussian) = d.μ
+median(d::GeneralizedGaussian) = d.μ
+mode(d::GeneralizedGaussian) = d.μ
+
+var(d::GeneralizedGaussian{T}) where {T<:Real} = (d.α^2) * ( gamma(3 / d.β) / gamma(1 / d.β) )
+std(d::GeneralizedGaussian{T}) where {T<:Real} = (d.α) * sqrt( gamma(3 / d.β) / gamma(1 / d.β) )
+
+skewness(d::GeneralizedGaussian{T}) where {T<:Real} = 0.0
+kurtosis(d::GeneralizedGaussian{T}) where {T<:Real} = (gamma(5 / d.β) * gamma(1 / d.β)) / (gamma(3 / d.β)^2) - 3
+
+entropy(d::GeneralizedGaussian{T}) where {T<:Real} = (1 / d.β) - log( d.β / (2 * d.α * gamma(1 / d.β) ) )
+
+#### Evaluation
+
+function pdf(d::GeneralizedGaussian{T}, x::Real) where T<:Real
+    (μ, α, β) = params(d)
+    return ( β / ( 2 * α * gamma(1 / β) ) ) * exp( -( abs(x - μ) / α )^β )
+end
+
+function cdf(d::GeneralizedGaussian{T}, x::Real) where T<:Real
+
+    #    To calculate the CDF, the incomplete gamma function is required. The CDF
+    #   of the Gamma distribution provides this, with the necessary 1/Γ(a) normalization.
+
+    (μ, α, β) = params(d)
+    return 1/2 + sign(x - μ) * ( cdf( Gamma( (1 / β), 1.0), (abs(x - μ) / α)^β) / 2 )
```

Comment:
Another thing I forgot to mention was that it would be good to avoid promotion when possible. Right now we probably don't support anything but `Float64` but once `Gamma` supports e.g. `Float32`s and `ForwardDiff.Dual`s the support would carry over here if we are careful. E.g. with something like
```suggestion
    v = cdf(Gamma(1 / β, 1), (abs(x - μ) / α)^β) / 2
    return typeof(v)(1/2) + sign(x - μ) * v
```
because `1` usually won't cause promotion in binary operations involving floating point numbers.

Reply:
That's a good consideration - should this change matriculate to the other functions as well? Namely the pdf and random sample generation functions

Reply:
> should this change matriculate to the other functions as well?

Yes but I think they are mostly fine. I've made two small suggestions.